### ISSUE TITLE
Azure Government Connections Getting No Query Results

### ISSUE THREAD

**Issue by ericmixon** [dataplat/dbatools#9006]
### Verified issue does not already exist?

I have searched and found no existing issue

### What error did you receive?

No error, but incorrect results.

### Steps to Reproduce

When setting up a connection to an Azure Government SQL Server.  I will first use Connect-AzAccount for a service principal and then request and AccessToken.  I then setup my DbaInstance connection like so.

`server = Connect-DbaInstance -SqlInstance "$SqlInstance" -AccessToken $token -Database $Database -AzureDomain $domain -NetworkProtocol TcpIp`

But when running a simple query. I get no results even though the same query in SSMS would return 1

`Invoke-DbaQuery -SqlInstance $server -Database $Database -Query "select value from sys.extended_properties where NAME = 'esms_version';"`

Using dbatools "Get-DbaExtendedProperty" function does the same thing as above.  I just get no results.

`$version = $(Get-DbaExtendedProperty -SqlInstance $server -Database $Database -Name esms_version).Value`

If I run the same commands against a local database, it works as expected, so this seems to only be an issue with my Azure Government connection. However, I can implement a solution with a SqlClient that works like so:

```
$connString = "Data Source=$SqlInstance; Initial Catalog=$Database; Encrypt=True; ";
$sqlConn = New-Object Microsoft.Data.SqlClient.SqlConnection
$sqlConn.AccessToken = $token.Token
$sqlConn.ConnectionString = $connString
$SqlQuery = "select value from sys.extended_properties where NAME = 'esms_version'"
$sqlConn.Open();
$sqlCommand = New-object Microsoft.Data.SqlClient.SqlCommand($sqlQuery,$sqlConn)
$sqlAdapter = New-Object Microsoft.Data.SqlClient.SqlDataAdapter($sqlCommand)
$ds = New-Object system.Data.DataSet
$version = $sqlAdapter.fill($ds)
```
This is using the same account credentials and configuration.  But gets the desired result.  Obviously though, it's way more complicated than we really want it to be. But it at least shows that it's not an issue with the account, authentication, network, or connection string parameters.

### Please confirm that you are running the most recent version of dbatools

2.0.3

### What PowerShell host was used when producing this error

Windows PowerShell (powershell.exe)

### PowerShell Host Version

Name                           Value                                                                                   
----                           -----                                                                                   
PSVersion                      5.1.19041.3031                                                                          
PSEdition                      Desktop                                                                                 
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                 
BuildVersion                   10.0.19041.3031                                                                         
CLRVersion                     4.0.30319.42000                                                                         
WSManStackVersion              3.0                                                                                     
PSRemotingProtocolVersion      2.3                                                                                     
SerializationVersion           1.1.0.1                                                                                 

### SQL Server Edition and Build number

Azure Sql Server

### .NET Framework Version

@{PSChildName=v2.0.50727; Version=2.0.50727.4927}
@{PSChildName=v3.0; Version=3.0.30729.4926}
@{PSChildName=Windows Communication Foundation; Version=3.0.4506.4926}
@{PSChildName=Windows Presentation Foundation; Version=3.0.6920.4902}
@{PSChildName=v3.5; Version=3.5.30729.4926}
@{PSChildName=Client; Version=4.8.04084}
@{PSChildName=Full; Version=4.8.04084}
@{PSChildName=Client; Version=4.0.0.0}

**Comment by wsmelton** [comment]
Does your `$server` object actually show its connected?

You can look at the verbose output to see what is being done to create the connection. We are creating the connection just as you do in Connect command:

https://github.com/dataplat/dbatools/blob/b1f2ba8327452f707608c312944a2cc896f45298/public/Connect-DbaInstance.ps1#L875-L901

I would predict it will be difficult for us to implement Azure Government cloud access because we (maintainers) do not have access to a tenant. When connecting to anything other than the Azure Cloud environment you have to specific which cloud you are connecting to, all of our code expects to be Azure Cloud. In order to support Azure Government we would (1) have to find where that cloud needs to be specified and (2) test against a real Gov tenant.

**Comment by ericmixon** [comment]
It's definitely connecting.  

I've done some more experimenting with Invoke-DbaQuery such as:

`Invoke-DbaQuery -SqlInstance $server -Database $Database -Query "select * from [sys].[extended_properties] where [NAME] = 'esms_version';"`

This returns results, but if I just try to select [value] from the query. I get nothing.

I've also noticed that I get similar behavior on other tables in our schema.  select * gives me expected results, but selecting specific columns gives me nothing.

I verify all the queries in SSMS to double check them.

Also: ` Get-DbaExtendedProperty -SqlInstance $server -Database $Database -Name esms_version` returns nothing.

**Comment by andreasjordan** [comment]
We don't do any magic with the query, so I can not explain the behavior.

Have you tried using `$server.ConnectionContext.ExecuteScalar("select value from sys.extended_properties where NAME = 'esms_version'")`? That would be a shorter workaround.

And maybe when running the command with `-Debug` it outputs some helpfull information.

**Comment by andreasjordan** [comment]
`Get-DbaExtendedProperty` does basically does a `$server.Databases[$Database].ExtendedProperties | Where-Object Name -eq esms_version`. Can you try using this?

**Comment by ericmixon** [comment]
I've realized now that the Get-DbaExtendedProperty was actually returning a connection error:

An error occurred: The following exception occurred while trying to enumerate the collection: "Failed to connect to server esms.database.usgovcloudapi.net.".

I'm guessing this is something to do with not supporting Azure Government.

Invoke-DbaQuery was returning results, but the object type changes based how I structure the query. That part is probably me just not being savvy with the innerworkings of SQL.

For now, I'm just going to write my own .net tool to replicate what I needed to do instead of using dbatools.  It was worth investigating though.  Maybe in the future, AzureGovernment can be supported.

Thank-you for your help and feedback!
